app/src/main/kotlin/codes/chrishorner/socketweather/home/HomeForecastPresenter.kt

```diff
@@ -27,6 +33,16 @@ class HomeForecastPresenter(view: View) {
     currentIcon.setImageDrawable(context.getWeatherIconFor(forecast.iconDescriptor, forecast.night))
     currentTemp.text = forecast.currentTemp.formatAsDegrees(context)
     feelsLikeTemp.text = forecast.tempFeelsLike?.formatAsDegrees(context) ?: "--"
+    humidity.text = forecast.humidity?.formatAsPercent(context) ?: "--"
+    wind.text = forecast.wind.run { "$speed_kilometre km/h $direction" }
+    uvProtection.text = forecast.todayForecast.uv.run {
+      if (start_time == null || end_time == null) {
+        "--"
+      } else {
+        val zoneId = ZoneId.systemDefault()
+        "${LocalTime.from(start_time.atZone(zoneId))}—${LocalTime.from(end_time.atZone(zoneId))}"
+      }
+    }
```

Comment:
I think this might look a little cleaner if we mark the view as `GONE` if we don't have UV times. Also we should make use of the time zone associated with the selected location, rather than the device's.

```suggestion
    forecast.todayForecast.uv.run {
      if (start_time == null || end_time == null) {
        uvProtection.isVisible = false
      } else {
        val zoneId = forecast.location.timezone
        uvProtection.text = "${start_time.atZone(zoneId).toLocalTime()}—${end_time.atZone(zoneId).toLocalTime()}"
      }
    }
```

Reply:
Agreed, done!